packages/eslint-plugin/src/rules/strict-enums.ts

```diff
@@ -0,0 +1,720 @@
+import * as util from '../util';
+import * as tsutils from 'tsutils';
+import * as ts from 'typescript';
+import { TSESTree } from '@typescript-eslint/utils';
+
+const TYPE_NAME_TRUNCATION_THRESHOLD = 40;
```

Comment:
I added this because it results in errors that are like 100 lines long with thousand of characters.
Like, the TypeScript compiler has a bunch of logic to neatly separate cascading type failures with indentation at each level, so I need some helper function to format the types like that. Does TypeScript expose something like that to help, or is there a util function in this monorepo that I can use?

Reply:
I would say just don't print the types at all then. None of the other rules do it.

Reply:
I'll remove it for now, but it's pretty bad UX, especially for more complicated types.

Just imagine if the TypeScript compiler did that. "This variable assignment is bugged, so I'll draw a red squiggly line next to it. But its up to you to figure out why." =p